### PR TITLE
refactor: Remove legacy social plugin, PDP, product grid and WYSIWYG code

### LINKED ISSUE #5393: Legacy PDP and product grid code are still there
## Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues)

## Issue Description
First of all, the Operator UI still has a Social section, which used to serve the purpose of adding social "share" buttons on the PDP, back in the 1.x days. This is not used anymore and this feature being present in Catalyst may confuse newcomers.

Second, lots of unused code relative to the legacy PDP, the product grid and the old WYSIWYG admin is still in this repo. We need to remove the unused parts, which will help reduce the client bundle size, improving both Catalyst and customer login UI loading times. Not to mention a small gain in DX as well.

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. Be a complete newcomer to Reaction, open Operator UI
2. Open the Social link under "Settings"
3. Try to understand how to use it
4. Fail at understanding because nothing is happening
5. Come to the conclusion that it's not used anymore
6. Get frustrated
7. Open the codebase
8. See lots of PDP components that are never rendered anywhere
9. Get frustrated again

## Possible Solution
Remove the unused code.

### PR DESCRIPTION
**Pull request by loan-laux**
Resolves #5393  
Impact: **major**  
Type: **performance|refactor**

## Issue
First of all, the Operator UI still has a Social section, which used to serve the purpose of adding social "share" buttons on the PDP, back in the 1.x days. This is not used anymore and this feature being present in Catalyst may confuse newcomers.

Second, lots of unused code relative to the legacy PDP, the product grid and the old WYSIWYG admin is still in this repo.

## Solution
We need to remove the unused parts, which will help reduce the client bundle size, improving both Catalyst and customer login UI loading times by a little bit. Not to mention a small gain in DX as well.

## Breaking changes
- The `social` plugin is no more.
- The following legacy components were removed:
  - `productDetail`
  - `ProductField`
  - `ProductTags`
  - `ProductMetadata`
  - `PriceRange`
  - `VariantList`
  - `SocialContainer`
  - `VariantListContainer`
  - `ProductNotFound`
- In the `default-theme` plugin, all the CSS files under the `products` directory were removed, with the exception of `products/variantForm.less` which still has literally 3 rules that are still being used.

## Testing
1. Open the Operator UI
2. Notice that everything works just as it used to
3. Notice that the client bundle is a bit lighter

Before:
Screen Shot 2019-07-31 at 15 32 26

After:
Screen Shot 2019-07-31 at 14 48 52

### CODE REVIEW

**Comment by kieckhafer** [comment]
Tgabks @loan-laux! Can you please take care of the DCO issues before I dive into this: https://github.com/reactioncommerce/reaction/runs/181723922

Thanks!

**Comment by machikoyasuda** [comment]
Actions to take after this PR is merged:
- Update the docs to remove any instructions for the Social plugin
- Update the docs to remove any old PDP/WYSIWIG component docs

File: imports/plugins/included/default-theme/client/styles/products/variantForm.less [hunk 1]
```diff
@@ -1,32 +1,4 @@
...
-  .form-group {
-
-  }
-  .checkbox,.radio {
-    margin-top: 0;
-    margin-bottom: 0;
-  }
-}
-
 .admin-controls-content .panel-default.variant-option-card > .panel-heading {
```
Comment (kieckhafer) [inline imports/plugins/included/default-theme/client/styles/products/variantForm.less]: @loan-laux where do you see these values being used? I don't see them being used anywhere, we could probably get rid of this entire file.

Comment (loan-laux) [reply imports/plugins/included/default-theme/client/styles/products/variantForm.less]: `admin-controls-content` is used in the `ActionView` component. Is that still being used though? I tried to dive into it but it's not really clear. Happy to delete if you can confirm. The less files the better!

Comment (loan-laux) [reply imports/plugins/included/default-theme/client/styles/products/variantForm.less]: Although the `ActionView` component still seems to be used (it's rendered directly in the `App` component and seems to be a core component of the routing system), I removed the CSS rules and no changes seemed to have happened. I guess a refactoring of this whole `ActionView` thing is in order, but that would be a separate and likely massive PR.

**Comment by loan-laux** [comment]
@kieckhafer Removed the `variantForm.less` file and fixed DCO.

**Pull request merged**